src/jit/lir.cpp

```diff
@@ -0,0 +1,1531 @@
+    assert(replacement != nullptr);
+    assert(IsDummyUse() || m_range->Contains(m_user));
+    assert(m_range->Contains(replacement));
+
+    GenTree* replacedNode = *m_edge;
+
+    *m_edge = replacement;
+    if (!IsDummyUse() && m_user->IsCall())
+    {
+        compiler->fgFixupArgTabEntryPtr(m_user, replacedNode, replacement);
+    }
+}
+
+//------------------------------------------------------------------------
+// LIR::Use::ReplaceWithLclVar: Assigns the def for this use to a local
+//                              var and points the use to a use of that
+//                              local var. If no local number is provided,
+//                              creates a new local var.
+//
+// For example, given the following IR:
+//
+//    t15 =    lclVar    int    arg1
+//    t16 =    lclVar    int    arg1
+//
+//          /--*  t15 int
+//          +--*  t16 int
+//    t17 = *  ==        int
+//
+//          /--*  t17 int
+//          *  jmpTrue   void
+//
+// If we wanted to replace the use of t17 with a use of a new local var
+// that holds the value represented by t17, we might do the following
+// (where `opEq` is a `Use` value that represents the use of t17):
+//
+//    opEq.ReplaceUseWithLclVar(compiler, block->getBBWeight(compiler));
+//
+// This would produce the following LIR:
+//
+//    t15 =    lclVar    int    arg1
+//    t16 =    lclVar    int    arg1
+//
+//          /--*  t15 int
+//          +--*  t16 int
+//    t17 = *  ==        int
+//
+//          /--*  t17 int
+//          *  st.lclVar int    tmp0
+//
+//    t18 =    lclVar    int    tmp0
+//
+//          /--*  t18 int
+//          *  jmpTrue   void
+//
+// Arguments:
+//    compiler - The Compiler context.
+//    blockWeight - The weight of the basic block that contains the use.
+//    lclNum - The local to use for temporary storage. If BAD_VAR_NUM (the
+//             default) is provided, this method will create and use a new
+//             local var.
+//
+// Return Value: The number of the local var used for temporary storage.
+//
+unsigned LIR::Use::ReplaceWithLclVar(Compiler* compiler, unsigned blockWeight, unsigned lclNum)
+{
+    assert(IsInitialized());
+    assert(compiler != nullptr);
+    assert(m_range->Contains(m_user));
+    assert(m_range->Contains(*m_edge));
+
+    GenTree* node = *m_edge;
+
+    if (lclNum == BAD_VAR_NUM)
+    {
+        lclNum = compiler->lvaGrabTemp(true DEBUGARG("ReplaceWithLclVar is creating a new local variable"));
+    }
+
+    // Increment its lvRefCnt and lvRefCntWtd twice, one for the def and one for the use
+    compiler->lvaTable[lclNum].incRefCnts(blockWeight, compiler);
+    compiler->lvaTable[lclNum].incRefCnts(blockWeight, compiler);
+
+    GenTreeLclVar* store = compiler->gtNewTempAssign(lclNum, node)->AsLclVar();
+    store->CopyCosts(node);
+
+    GenTree* load =
+        new (compiler, GT_LCL_VAR) GenTreeLclVar(store->TypeGet(), store->AsLclVarCommon()->GetLclNum(), BAD_IL_OFFSET);
```

Comment:
This may not matter, given how poorly we maintain the IL offsets, but I would think it would be appropriate to use the IL offset of the node being replaced.